gobblin-api/src/main/java/org/apache/gobblin/dataset/DatasetDescriptor.java

```diff
@@ -17,34 +17,52 @@
 
 package org.apache.gobblin.dataset;
 
-import java.util.Map;
-
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
-
 import lombok.Getter;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Objects;
+
 
 /**
  * A {@link Descriptor} identifies and provides metadata to describe a dataset
  */
 public class DatasetDescriptor extends Descriptor {
   private static final String PLATFORM_KEY = "platform";
   private static final String NAME_KEY = "name";
+  private static final String CLUSTER_NAME_KEY = "clusterName";
 
   /**
    * which platform the dataset is stored, for example: local, hdfs, oracle, mysql, kafka
    */
   @Getter
   private final String platform;
 
+  /**
+   * Human-readeable cluster name.
+   *
+   * @see org.apache.gobblin.util.ClustersNames
+   */
+  @Getter
+  @Nullable
+  private final String clusterName;
```

Comment:
What exactly does clusterName mean? Is this the name of the storage instance that has an instance of a given dataset? If so, I think storageInstance or storageInstanceName would better capture what this field is.

Reply:
What is the expected format for the clusterName? Can it be any string? I think it would be better to define a new class called "StorageIdentifier" that includes among other things fields such as logicalName, address, port, availability zone, etc.  We should also define a URN representation for the storage instance.

Reply:
Initially, I was thinking about putting a well-known cluster name like "snap" into this field, and for unknown clusters put a DNS name like "datalake.corp.com". Then lineage events will include this name as-is, and it can be displayed to users directly.  We already use "cluster name" concept in org.apache.gobblin.util.ClustersNames, for getting nice names from yarn/hdfs urls.

However, this well-know name is not easily convertible back to domain name. Single well-known name can map to different urls for hdfs/hive. So, there could be a benefit in keeping a hostname instead of clustername, and converting it to human-readable name before reporting lineage. 

DatasetDescriptor already looks like a dataset URL with non-standard field names:
- platform is like url.protocol - "hdfs", "kafka", "mysql"
- name is like url.path - "table1" or "/data/tracking/test123"
- clusterName is like a url.host - "my-cluster" or "my-cluster.company.com"

Regarding the fields you suggested:
- I don't think we have availability zone information anywhere.
- We can get well-known nice name from hostname using org.apache.gobblin.util.ClustersNames , so don't have to store it separately.
- Keeping port information is discussable. I would assume that any production deployments will have separate DNS names for different clusters. However, for small scale, temporary deployments it's possible to have multiple Hives or namenodes on a single host, running on different ports.